### ISSUE TITLE
Trainer Evaluate not working

### ISSUE THREAD

**Issue by Sharad24** [SforAiDl/genrl#321]
Probably as a result of #294 

Issue might be that there is a `detach().clone()` missing somewhere. What ends up happening is that an underlying pointer reference is passes on which later becomes 0. For e.g.

```python
rewards = torch.zeros(10) # Assuming there are 10 envs
rewards[i] = self.env.rewards[i] # This might be incompatible. As a clone operation might be needed.
```

Similarly for states, actions, etc
